Neos.ContentRepository.Core/Classes/Dimension/ContentDimensionValueVariationEdges.php

```diff
@@ -17,50 +17,32 @@
 /**
  * A set of content dimension value variation edges
  *
- * @implements \IteratorAggregate<int,ContentDimensionValueVariationEdge>
+ * @implements \IteratorAggregate<ContentDimensionValueVariationEdge>
  * @internal
  */
 final class ContentDimensionValueVariationEdges implements \IteratorAggregate
 {
     /**
-     * @var array<int,ContentDimensionValueVariationEdge>
+     * @var array<ContentDimensionValueVariationEdge>
      */
     private array $edges;
 
-    /**
-     * @var \ArrayIterator<int,ContentDimensionValueVariationEdge>
-     */
-    private \ArrayIterator $iterator;
-
-    /**
-     * @param array<int,ContentDimensionValueVariationEdge> $array
-     */
-    public function __construct(array $array)
+    public function __construct(ContentDimensionValueVariationEdge ...$edges)
     {
-        foreach ($array as $edge) {
-            if (!$edge instanceof ContentDimensionValueVariationEdge) {
-                throw new \InvalidArgumentException(
-                    'ContentDimensionValueVariationEdges may only contain ContentDimensionValueVariationEdge objects',
-                    1639661280
-                );
-            }
-        }
-
-        $this->edges = $array;
-        $this->iterator = new \ArrayIterator($this->edges);
+        $this->edges = $edges;
     }
 
     public static function createEmpty(): self
     {
-        return new self([]);
+        return new self();
     }
 
     /**
-     * @return \ArrayIterator<int,ContentDimensionValueVariationEdge>|ContentDimensionValueVariationEdge[]
+     * @return \Traversable<int,ContentDimensionValueVariationEdge>
      */
-    public function getIterator(): \ArrayIterator
+    public function getIterator(): \Traversable
     {
-        return $this->iterator;
+        return new \ArrayIterator($this->edges);
```

Comment:
probably less expensive, and also a subtype of `\Traversable` would be to use:

```php
yield from $this->edges;
```

if we agree on this im happy to regex replace the usages ^^

Reply:
did it